pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -335,6 +335,18 @@ func ValidateACMEChallengeSolverDNS01(p *cmacme.ACMEChallengeSolverDNS01, fldPat
 				if len(p.AzureDNS.TenantID) == 0 {
 					el = append(el, field.Required(fldPath.Child("azureDNS", "tenantID"), ""))
 				}
+				if len(p.AzureDNS.ManagedIdentityClientID) > 0 {
+					el = append(el, field.Forbidden(fldPath.Child("azureDNS", "managedIdentityClientID"), "managed identity can not be used at the same time as clientID, tenantID or clientSecret"))
+				}
+				if len(p.AzureDNS.ManagedIdentityResourceID) > 0 {
+					el = append(el, field.Forbidden(fldPath.Child("azureDNS", "managedIdentityResourceID"), "managed identity can not be used at the same time as clientID, tenantID or clientSecret"))
+				}
```

Comment:
Should these restrictions also be reflected in the API docs?

Reply:
Yes, it needs to be clearly specified somewhere that you can either use a service principal (client id and client secret) or managed identity (with optional client or resource id) but never both. Where would be the best place to put this? If I move the managed identity ids to a nested field, maybe that would be a good place for it?

Reply:
please let me know if you like the state of validation after the new commmits